text/trace/0205-messaging-semantic-conventions-context-propagation.md

```diff
@@ -0,0 +1,177 @@
+consumer.
+
+However, this is not a trivial assumption, and it is not easily accommodated by
+existing established context propagation mechanisms. Those mechanisms propagate
+context on a per-request basis, whereas the messaging semantic conventions
+assume that context is propagated on a per-message basis. This means, that
+although several requests might be involved in the processing of a single
+message (publishing a message, fetching a message, potentially multiple times by
+multiple consumers), it is assumed that all components have access to the same
+per-message context information that allows correlating all the stages of
+processing a message.
+
+To achieve this desired outcome, a context needs to be attached to a message,
+and intermediaries must not alter the context attached to the message. _This
+requirement should be documented, as it is an important factor in deciding how
+to propagate context for message scenarios and how to standardize context
+propagation for existing message protocols._
+
+The additions proposed in this document neither break nor invalidate any of
+the existing semantic conventions for messaging, but rather make an implicit
+requirement explicit.
+
+### Example
+
+Many intermediaries (message brokers) offer REST APIs for publishing and
+fetching messages. A producer can publish a message by sending an HTTP request
+to the intermediary and a consumer can pull a message by sending an HTTP request
+to the intermediary:
+
+```
+  +----------+
+  | Producer |
+  +----------+
+       |
+       | HTTP POST (publishing a message)
+       v
++--------------+
+| Intermediary |
++--------------+
+       ^
+       | HTTP GET (fetching a message)
+       |
+  +----------+
+  | Consumer |
+  +----------+
+```
+
+Existing semantic conventions suppose that the consumer can use context
+information from the producer trace to create links or parent/child
+relationships between consumer and producer traces. For this to be possible,
+context information from the producer needs to be propagated to the consumer.
+In the example outlined above, the consumers sends an HTTP GET request to the
+intermediary to fetch a message, the message is returned as part of the
+response. Via this HTTP request, context information can be propagated from the
+consumer to the intermediary, but not from the intermediary to the consumer.
+The consumer can obtain the necessary producer context information only if it
+is propagated as part of the message itself, independent of HTTP context
+propagation.
+
+For correlating producer and consumer traces without special intermediary
+instrumentation it is thus necessary to attach a producer context to the
+message so it can be extracted and used by the consumer, regardless of the
+contexts that are propagated on HTTP requests for publishing and fetching the
+message.
+
+Although OpenTelemetry semantic conventions cannot specify the exact mechanisms
```

Comment:
When adding this to the Specification, concise examples (say, "this is how Kafka supports this") will be really nice to have.